deeplab/demo/src/index.js

```diff
@@ -0,0 +1,172 @@
+  legendContainer.classList.add('is-invisible');
+  const image = document.getElementById('input-image');
+  image.src = src;
+  const imageContainer = document.getElementById('input-card');
+  imageContainer.classList.remove('is-invisible');
+  status('Waiting until the model is picked...');
+};
+
+const processImage = file => {
+  if (!file.type.match('image.*')) {
+    return;
+  }
+  const reader = new FileReader();
+  reader.onload = event => {
+    setImage(event.target.result);
+  };
+  reader.readAsDataURL(file);
+};
+
+const processImages = event => {
+  const files = event.target.files;
+  Array.from(files).forEach(processImage);
+};
+
+const displaySegmentationMap = deeplabOutput => {
+  const [legend, height, width, segmentationMap] = deeplabOutput;
+  const canvas = document.getElementById('output-image');
+  const ctx = canvas.getContext('2d');
+
+  const outputContainer = document.getElementById('output-card');
+  outputContainer.classList.remove('is-invisible');
+  const segmentationMapData = new ImageData(segmentationMap, width, height);
+  canvas.style.width = '100%';
+  canvas.style.height = '100%';
+  canvas.width = width;
+  canvas.height = height;
+  ctx.putImageData(segmentationMapData, 0, 0);
+
+  const legendList = document.getElementById('legend');
+  while (legendList.firstChild) {
+    legendList.removeChild(legendList.firstChild);
+  }
+
+  // Work around the parcel failure with Object.keys
+  for (const label in legend) {
+    if (legend.hasOwnProperty(label)) {
+      const tag = document.createElement('span');
+      tag.innerHTML = label;
+      const [red, green, blue] = legend[label];
+      tag.classList.add('column');
+      tag.style.backgroundColor = `rgb(${red}, ${green}, ${blue})`;
+      tag.style.padding = '1em';
+      tag.style.margin = '1em';
+      tag.style.color = '#ffffff';
+
+      legendList.appendChild(tag);
+    }
+  }
+
+  const legendContainer = document.getElementById('legend-card');
+  legendContainer.classList.remove('is-invisible');
+};
+
+const sleep = ms => {
+  return new Promise(resolve => setTimeout(resolve, ms));
+};
+
+const status = message => {
+  const statusMessage = document.getElementById('status-message');
+  statusMessage.innerText = message;
+};
+
+const runDeeplab = async modelName => {
+  status(`Running the inference...`);
+  await sleep(100);
+  const initialisationStart = performance.now();
+  const isQuantizationDisabled = document.getElementById(
+    'is-quantization-disabled'
+  ).checked;
+  if (!(isQuantizationDisabled ^ state.isQuantized)) {
+    state.isQuantized = !isQuantizationDisabled;
+    initialiseModels();
+    sleep(100);
+  }
+  const input = document.getElementById('input-image');
+  if (!input.src || !input.src.length || input.src.length === 0) {
+    status('Failed! Please load an image first.');
+    return;
+  }
+
+  const model = deeplab[modelName];
+  if (input.complete && input.naturalHeight !== 0) {
```

Comment:
Seems like repetitive code here. Can we move prediction logic to separate function?